include/pipettes/core/eeprom.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <array>
+#include <concepts>
+#include <cstdint>
+
+#include "common/core/bit_utils.hpp"
+
+/*
+ * A template and helper functions describing
+ */
+
+namespace i2c {
+template <class I2C>
+concept EEPromPolicy =
+    requires(I2C i2c_comms, const uint8_t transmit,
```

Comment:
shouldn't this allow a larger buffer for the serial to write? i guess it doesn't matter much right now but It might save us some pain down the road

Reply:
The HAL Transmit function only allows 8 bits of data to be sent at a time. If we want to write more, we would probably need to implement something in the individual write function that takes in a buffer and parses that buffer appropriately. I think we should keep it as uint8_t for now until we have a better idea of the serial number size we'll be sending.